lambdas/handlers/create_document_reference_handler.py

```diff
@@ -19,23 +20,42 @@
 def lambda_handler(event, context):
     logger.info("Starting document reference creation process")
 
+    try:
+        document_type = (event["queryStringParameters"]["documentType"]).upper()
+        if document_type in SupportedDocumentTypes.list_names():
+            logger.info("Provided document is supported")
+        else:
+            raise Exception("Provided document type is not supported")   
+    except Exception as e:
+        logger.error(e)
+        response = ApiGatewayResponse(400, "An error occured processing the required document type", "POST").create_api_gateway_response()
+        return response
+    
+    # We cannot get here unless a valid document type has been provided, therefore we can default to ARF
     try:
         s3_bucket_name = os.environ["DOCUMENT_STORE_BUCKET_NAME"]
         dynamo_table = os.environ["DOCUMENT_STORE_DYNAMODB_NAME"]
 
-        body = json.loads(event["body"])
+        if (document_type == SupportedDocumentTypes.LG.name):
```

Comment:
Might be more readable if this was something like
```
if document_type == "LG":
    s3_bucket_name = os.environ["LG_BUCKET"]
else:
    s3_bucket_name = os.environ["ARF_BUCKET"]

```
should work just fine, as Python is a bit looser in the scoping than Java so you don't need to declare the variable before the if statement